### DISCUSSION TITLE
Failure to compile

### DISCUSSION THREAD

**Discussion by SchroedingersChinchilla** [grimme-lab/xtb#706] [Q&A]
Hi all,

I'm working on my university's computing cluster, to which I connect remotely in a Linux environment.  I'm trying to get xtb running in our community projects directory, such that I can use it with ORCA 5.0.3, which is also installed there.  I loaded (as modules) CMake 3.19.5 and the Intel compiler 19.1.1.  I downloaded xtb-6.5.1-source.tar.xz and unpacked it in the destination folder.  As per the readme file, I issued the following command:

cmake -B build DCMAKE_BUILD_TYPE=Release
make -C build

Everything was fine until this step was ~90% done, and then I got a bunch of error messages, as follows:

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(122): error: expected an expression
    if (!check(energy, -8.38393864716134, 1.0e-9, "Energy does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(122): error: expected an expression
    if (!check(energy, -8.38393864716134, 1.0e-9, "Energy does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(122): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    if (!check(energy, -8.38393864716134, 1.0e-9, "Energy does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(124): error: expected an expression
    if (!check(q[5], 0.06090868805034, 1.0e-8, "Charge does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(124): error: expected an expression
    if (!check(q[5], 0.06090868805034, 1.0e-8, "Charge does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(124): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    if (!check(q[5], 0.06090868805034, 1.0e-8, "Charge does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(126): error: expected an expression
    if (!check(dipole[2], -0.35455233974705, 1.0e-6, "Dipole does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(126): error: expected an expression
    if (!check(dipole[2], -0.35455233974705, 1.0e-6, "Dipole does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(126): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    if (!check(dipole[2], -0.35455233974705, 1.0e-6, "Dipole does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(128): error: expected an expression
    if (!check(wbo[9], +2.89453979224265, 1.0e-8, "Bond order does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(128): error: expected an expression
    if (!check(wbo[9], +2.89453979224265, 1.0e-8, "Bond order does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(128): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    if (!check(wbo[9], +2.89453979224265, 1.0e-8, "Bond order does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(131): error: type name is not allowed
    xtb_delete(res);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(131): error: expected a ")"
    xtb_delete(res);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(131): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    xtb_delete(res);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(132): error: type name is not allowed
    xtb_delete(calc);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(132): error: expected a ")"
    xtb_delete(calc);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(132): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    xtb_delete(calc);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(133): error: type name is not allowed
    xtb_delete(mol);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(133): error: expected a ")"
    xtb_delete(mol);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(133): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    xtb_delete(mol);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(134): error: type name is not allowed
    xtb_delete(env);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(134): error: expected a ")"
    xtb_delete(env);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(134): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    xtb_delete(env);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(141): error: expected an expression
    if (!check(tester, 1, "Results not deleted"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(141): error: expected an expression
    if (!check(tester, 1, "Results not deleted"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(141): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    if (!check(tester, 1, "Results not deleted"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(144): error: expected an expression
    if (!check(tester, 1, "Calculator not deleted"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(144): error: expected an expression
    if (!check(tester, 1, "Calculator not deleted"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(144): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    if (!check(tester, 1, "Calculator not deleted"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(147): error: expected an expression
    if (!check(tester, 1, "Molecule not deleted"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(147): error: expected an expression
    if (!check(tester, 1, "Molecule not deleted"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(147): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    if (!check(tester, 1, "Molecule not deleted"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(150): error: expected an expression
    if (!check(tester, 1, "Environment not deleted"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(150): error: expected an expression
    if (!check(tester, 1, "Environment not deleted"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(150): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    if (!check(tester, 1, "Environment not deleted"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(157): error: type name is not allowed
    xtb_delete(res);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(157): error: expected a ")"
    xtb_delete(res);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(157): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    xtb_delete(res);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(158): error: type name is not allowed
    xtb_delete(calc);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(158): error: expected a ")"
    xtb_delete(calc);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(158): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    xtb_delete(calc);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(159): error: type name is not allowed
    xtb_delete(mol);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(159): error: expected a ")"
    xtb_delete(mol);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(159): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    xtb_delete(mol);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(160): error: type name is not allowed
    xtb_delete(env);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(160): error: expected a ")"
    xtb_delete(env);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(160): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    xtb_delete(env);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(271): error: type name is not allowed
    xtb_delete(res);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(271): error: expected a ")"
    xtb_delete(res);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(271): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    xtb_delete(res);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(272): error: type name is not allowed
    xtb_delete(calc);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(272): error: expected a ")"
    xtb_delete(calc);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(272): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    xtb_delete(calc);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(273): error: type name is not allowed
    xtb_delete(mol);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(273): error: expected a ")"
    xtb_delete(mol);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(273): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    xtb_delete(mol);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(274): error: type name is not allowed
    xtb_delete(env);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(274): error: expected a ")"
    xtb_delete(env);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(274): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    xtb_delete(env);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(276): error: expected an expression
    if (!check(pcgrad[0], 0.00000755, 1.0e-6, "pcgrad[0] does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(276): error: expected an expression
    if (!check(pcgrad[0], 0.00000755, 1.0e-6, "pcgrad[0] does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(276): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    if (!check(pcgrad[0], 0.00000755, 1.0e-6, "pcgrad[0] does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(278): error: expected an expression
    if (!check(pcgrad[95], 0.00001312, 1.0e-6, "pcgrad[95] does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(278): error: expected an expression
    if (!check(pcgrad[95], 0.00001312, 1.0e-6, "pcgrad[95] does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(278): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    if (!check(pcgrad[95], 0.00001312, 1.0e-6, "pcgrad[95] does not match"))
         ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(285): error: type name is not allowed
    xtb_delete(res);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(285): error: expected a ")"
    xtb_delete(res);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(285): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    xtb_delete(res);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(286): error: type name is not allowed
    xtb_delete(calc);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(286): error: expected a ")"
    xtb_delete(calc);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(286): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    xtb_delete(calc);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(287): error: type name is not allowed
    xtb_delete(mol);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(287): error: expected a ")"
    xtb_delete(mol);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(287): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    xtb_delete(mol);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(288): error: type name is not allowed
    xtb_delete(env);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(288): error: expected a ")"
    xtb_delete(env);
    ^

/projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c(288): error: expression preceding parentheses of apparent call must have (pointer-to-) function type
    xtb_delete(env);
    ^

compilation aborted for /projects/community/xtb/6.5.1/wpk25/xtb-6.5.1/test/api/c_api_example.c (code 2)
make[2]: *** [test/api/CMakeFiles/xtb_tests_capi.dir/c_api_example.c.o] Error 2
make[2]: Leaving directory `/projectsp/community/xtb/6.5.1/wpk25/xtb-6.5.1/build'
make[1]: *** [test/api/CMakeFiles/xtb_tests_capi.dir/all] Error 2
make[1]: Leaving directory `/projectsp/community/xtb/6.5.1/wpk25/xtb-6.5.1/build'
make: *** [all] Error 2
make: Leaving directory `/projectsp/community/xtb/6.5.1/wpk25/xtb-6.5.1/build'

Any ideas in how I can resolve this?

Thanks!

**Comment by cplett** [comment]
Hi, 
This is probably a compiler problem. We tried this out with different compiler versions starting with 2021.2.0 and this worked out without errors. If you have the possibility, I would recommend using a newer compiler version.

**Reply by SchroedingersChinchilla** [reply]
Thanks, I'll give it a shot!
